pkg/trace/writer/trace.go

```diff
@@ -216,8 +216,25 @@ func (w *TraceWriter) flush() {
 		gzipw.Write(b)
 		gzipw.Close()
 
-		for _, sender := range w.senders {
-			sender.Push(p)
+		if len(w.senders) == 1 {
+			// Avoid an allocation when only one endpoint is configured.
```

Comment:
It isn't clear when reading the code which allocation we are avoiding (unless we read it starting at the bottom). For that reason, can the comment just be:
```go
// fast path
```
This is a common comment style throughout the Go repository. It usually signifies that this is the fast, clear and easiest path. The second (non-fast) path is the one warranting explanations.